src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -143,7 +143,7 @@ private async Task AnalyzeProjectAsync(Project project, bool forceAnalyzerRun, C
                 // this is perf optimization. we cache these result since we know the result. (no diagnostics)
                 var activeAnalyzers = stateSets
                                         .Select(s => s.Analyzer)
-                                        .Where(a => DocumentAnalysisExecutor.IsAnalyzerEnabledForProject(a, project, GlobalOptions) && !a.IsOpenFileOnly(ideOptions.CleanupOptions?.SimplifierOptions));
+                                        .Where(a => (forceAnalyzerRun || DocumentAnalysisExecutor.IsAnalyzerEnabledForProject(a, project, GlobalOptions)) && !a.IsOpenFileOnly(ideOptions.CleanupOptions?.SimplifierOptions));
```

Comment:
interesting, was this a bug independent of this PR?

Reply:
Yes, correct. This fixes the separate issue that Sam mentioned in the teams channel when user has completely turned off all background analysis (set to `None`, not even current document or open documents) and then they execute `Run Code Analysis` command.